doc/guides/misc/how_to_doc.md

```diff
@@ -0,0 +1,156 @@
+
+Compared to the Doxygen output we want to be extremely careful about the navigation and only show the guides that are relevant to the user and in a logical order that does not overwhelm the user with too many options at once.
+
+Generally, the less information you declare within the config the better, given that ideally your frontmatter information is enough to generate the navigation.
+
+You can either add your guide to an existing section or create a new section. In the event that you want to simply append your guide to an existing section, you can simply add the file name to the list of files in the section. The root of the so called `slug` of your file is the guides folder itself, so a guide under `doc/guides/test/hello_world.md` will have the slug `test/hello_world.md`.
+
+:::note
+There is one exception to this rule, if your file is called index.md, the slug will be the name of the folder it is in. For example `doc/guides/test/index.md` will have the slug `test`.
+:::
+
+### Creating a new section
+
+Generally, we want to avoid creating new sections unless absolutely necessary. For example, if you create a totally new tutorial that spans multiple files, it might make sense to create a new section.
+In this case, you can create a new section by adding a new entry to the `sidebar` array in the `astro.config.mjs` file. The entry should look like this:
+
+```javascript title="Adding a new section"
+sidebar: [
+    {
+        label: "The Best Tutorial",
+        items: [
+            "tutorial/best_tutorial",
+        ],
+    },
+],
+```
+
+:::tip
+Sections can also be nested, so you can create a section within a section by adding another array to the `items` array. This is useful for organizing your guides into sub-sections.
+:::
+
+### Code Block Testing
+
+If you are writing a guide that includes code blocks it is important to test them to make sure they work as expected.
+To facilitate this, we have a small script that runs over the code blocks in the guide and compares them to a specified example folder.
+
+For example, let's say that we have a tutorial that explains the hello world example.
+This example is located in `examples/basic/hello-world/` and the guide is located in `doc/guides/test/hello_world.md`.
+
+First we have to tell the script where to find the example folder. This is done by adding the `code_folder` key to the frontmatter of the guide. The value should be the path to the example folder relative to the root of the repository.
+
+```markdown title="Example guide with code_folder"
+---
+title: Tutorial: Hello World
+description: This is a description of my new guide
+code_folder: examples/basic/hello-world/
+---
+```
+
+Then we can add the code blocks to the guide. The script will automatically detect the code blocks and run them against the example folder.
+
+```c title="This code block will be tested against the hello world example"
+int main(void)
+{
+    puts("Hello World!");
+```
```

Comment:
`````suggestion
Then we can add the code blocks to the guide. The script will automatically detect the code blocks and run them against the example folder.

````markdown  title="This code block will be tested against the hello world example"
```c
int main(void)
{
    puts("Hello World!");
```
````
`````

Reply:
ty